app/models/katello/concerns/hostgroup_extensions.rb

```diff
@@ -5,35 +5,36 @@ module HostgroupExtensions
 
       included do
         before_save :add_organization_for_environment
-        belongs_to :kickstart_repository, :class_name => "::Katello::Repository",
-                   :foreign_key => :kickstart_repository_id, :inverse_of => :kickstart_hostgroups
-        belongs_to :content_source, :class_name => "::SmartProxy", :foreign_key => :content_source_id, :inverse_of => :hostgroups
-        belongs_to :content_view, :inverse_of => :hostgroups, :class_name => "::Katello::ContentView"
-        belongs_to :lifecycle_environment, :inverse_of => :hostgroups, :class_name => "::Katello::KTEnvironment"
-
-        validates_with Katello::Validators::ContentViewEnvironmentValidator
-        validates_with Katello::Validators::HostgroupKickstartRepositoryValidator
-        validates_with ::AssociationExistsValidator, attributes: [:content_source]
+        has_one :kickstart_repository, :through => :content_facet
```

Comment:
I think an association is missing

```
[8] pry(main)> Hostgroup.first.content_source
ActiveRecord::HasManyThroughAssociationNotFoundError: Could not find the association :content_facet in model Hostgroup
```

Reply:
@ShimShtein is this user error on my part? :)

Reply:
@jturel, sounds like something on your part... Can you try and `spring stop` before you try again? It feels like a reloading problem.